src/openapi-generator.ts

```diff
@@ -345,6 +354,11 @@ export class OpenAPIGenerator {
       : simpleSchema;
   }
 
+  private generateSecuritySchema(name: string, schema: SecuritySchemeObject): SecuritySchemeObject {
+    this.securitySchemaRefs[name] = schema;
```

Comment:
From what I see you're aiming to provide a pure object definition for the `components.securitySchemes` property, right? If that is the case I see no reason for putting this through the registry and binding it into the generation logic. I'll try and showcase a different approach.
```ts
// Note I think this would look better in a class/factory but a function is easier to shwocase. We can just export this as
// some sort of helper function from `zod-to-openapi`.
function generateSecuritySchemes<T extends Record<string, SecuritySchemeObject>>(securitySchemes: T) {
  return {
    securitySchemes,
    addSecurityScheme: (key: keyof T, val: string[] = []) => ({[key]: val}),
  }
}


  const generator = new OpenAPIGenerator(rootRegistry.definitions);

  const Security = generateSecuritySchemes({bearerAuth: {
    type: 'http',
    scheme: 'bearer',
    bearerFormat: 'JWT',
  }})

  registry.registerPath({
    path: '/units',
    method: 'get',
    security: [
      Security.addSecurityScheme('bearerAuth'),
    ],
    responses: {
      200: {
        mediaType: 'application/json',
        schema: Unit.array().openapi({description: 'Array of all units'})
      }
    }
  })

  generator.generateDocument({
    // the generic stuff
    components: {
      securitySchemes: Security.securitySchemes
    }
  });
```

In order to make this work we would only need to make sure the config is being merged correctly in `generateDocument`. I feel like it is not wrong to provide support for manually added components (if someone wants to do that).

@viktor-ku @georgyangelov what do you think. I'd also love to here any possible benefits of getting it through the registry, since I wrote this on the fly and haven't tested particular cases.

Reply:
It's certainly reasonable, however I think that _registry_ could serve a bit more than just a zod schema handler. The _registry_ just sounds like a convenient way to just add stuff to it like security schemas and it would just know what to do with it.

Maybe it's better to rename it from `registry.registerSecuritySchema` to like `registry.addSecuritySchema` implying that there is nothing to generate, just _add_ to the registry?

And then I don't believe it's necessary to have additional boilerplate around `generateDocument`, I believe the registry is therefore capable of handling exactly that: knowing when to add something to the components and then doing it.